lib/build/build.js

```diff
@@ -85,23 +87,14 @@ class Build {
       }
 
       if (peonConfig.cache && peonConfig.cache.length) {
-        await this._runStep('save cache', () => this._saveCache())
+        await this._runStep('save cache', (updateOutput) =>
+          this._saveCache(updateOutput)
+        )
       }
 
-      await this._runStep('deploy', () => this._deploy())
-
-      let {
-        destination: { absoluteUrl },
-        pathInDestination
-      } = this
+      let deployData = await this._runStep('deploy', () => this._deploy())
 
-      // We cannot use path.join here because of the protocol:// prefix
-      if (!absoluteUrl.endsWith('/')) {
-        absoluteUrl = `${absoluteUrl}/`
-      }
-      await status.finishBuild(buildId, 'success', {
-        outputURL: `${absoluteUrl}${env.evaluate(pathInDestination)}`
-      })
+      await status.finishBuild(buildId, 'success', deployData)
```

Comment:
All the destination path/url stuff was moved to the deploy step, which now returns it. Here we just pass what deploy returned to be saved in the job extra data.